python/Ganga/new_tests/GPI/GangaUnitTest.py

```diff
@@ -82,8 +83,12 @@ def start_ganga():
 
     # [PollThread]autostart_monThreads=False has turned this off being done automatically.
     # The thread pool is emptied by _ganga_run_exitfuncs
-    from Ganga.Core.MonitoringComponent.Local_GangaMC_Service import _makeThreadPool
-    _makeThreadPool()
+    from Ganga.Core.MonitoringComponent.Local_GangaMC_Service import ThreadPool, _makeThreadPool
+    if len(ThreadPool) > 0:
+        del ThreadPool[:]
+        ThreadPool = []
```

Comment:
The _makeThreadPool was never intended to wipe/clear the ThreadPool only to append to it, which is what concerned me about it's behaviour.
This seemed to work, although I can just as easily change the method to reset the thread pool list rather than raise an exception which it now does.